### PR TITLE
Fixed droppers consistently saying 'You can't do that, the dropper is not open."

### PR DESCRIPTION
**Pull request by twirling-jesters**
Science has come very far. Droppers will no longer give a "You can't do that, The Dropper isn't open" whenever you try to squirt something into someone's eyes.
dreamseeker_2020-07-15_23-52-44

:cl:
 * bugfix: Droppers no longer give redundant/useless messages about not being able to open them.

### CODE REVIEW

**Comment by DamianX** [comment]
You've correctly identified the source of the problem but your solution is not optimal.
Type-checks (`istype()`) on `src` are always a code smell, you can rewrite it in a better way.

The root cause of the issue is that dropper behavior has been implemented in `afterattack()`, for some unknown reason.
Instead of adding a check inside `/obj/item/weapon/reagent_containers/attack`, simply implement `/obj/item/weapon/reagent_containers/dropper/attack` by moving the logic that's currently inside `afterattack` into it.

**Comment by twirling-jesters** [comment]
Sorry for the multiple (probably confusing) commits, I'm still getting used to git. And yeah, the dropper file seemed to have a lot of bad practice in there (checking after the attack if what was attacked was a mob, instead of just having a proc for attacking a mob exclusively). I may look into doing some housekeeping on some of the reagent containers after this.

File: code/modules/reagents/reagent_containers/dropper.dm [hunk 1]
```diff
@@ -22,6 +22,32 @@
 /obj/item/weapon/reagent_containers/dropper/update_icon()
 	icon_state = "dropper[(reagents.total_volume ? 1 : 0)]"
 
+/obj/item/weapon/reagent_containers/dropper/attack(mob/M as mob, mob/user as mob)
```
Comment (DamianX) [inline code/modules/reagents/reagent_containers/dropper.dm:25]:
```suggestion
/obj/item/weapon/reagent_containers/dropper/attack(mob/M, mob/user)
```

File: code/modules/reagents/reagent_containers/dropper.dm [hunk 2]
```diff
@@ -22,6 +22,32 @@
...
+			user.visible_message("<span class='danger'>[user] tries to squirt something into [M]'s eyes, but fails!</span>")
+			src.reagents.reaction(safe_thing, TOUCH)
+			src.reagents.remove_any(amount_per_transfer_from_this)
+			update_icon()
+			return
+	user.visible_message("<span class='danger'>[user] squirts something into [M]'s eyes!</span>")
+	src.reagents.reaction(M, TOUCH)
+	src.reagents.remove_any(amount_per_transfer_from_this)
+	update_icon()
+	return
```
Comment (DamianX) [inline code/modules/reagents/reagent_containers/dropper.dm:49]: return at the end of a proc does nothing, remove it

**Pull request merged**